service/src/test/java/greencity/filters/HabitFactSpecificationTest.java

```diff
@@ -0,0 +1,236 @@
+package greencity.filters;
+
+import greencity.entity.*;
+import jakarta.persistence.criteria.*;
+import jakarta.persistence.metamodel.SingularAttribute;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+
+@ExtendWith(MockitoExtension.class)
+class HabitFactSpecificationTest {
+
+    @Mock
+    private Root<HabitFact> HabitFactRoot;
```

Comment:
fields should be in camelCase like rabbit suggested
```suggestion
    @Mock
    private Root<HabitFact> habitFactRoot;
```